cfg/builder/builder_walk.cc

```diff
@@ -71,6 +99,12 @@ LocalRef unresolvedIdent2Local(CFGContext cctx, const ast::UnresolvedIdent &id)
         case ast::UnresolvedIdent::Kind::Instance:
             ENFORCE(cctx.ctx.owner.isMethod());
             klass = cctx.ctx.owner.owner(cctx.ctx).asClassOrModuleRef();
+
+            if (cctx.link != nullptr) {
+                // We're inside a block, so we may have to change the receiver to the proc's binding
+                klass = resolveBlockContext(cctx, klass);
+            }
```

Comment:
There's no reason to treat instance variables specially here; class variables should get the same treatment, shouldn't they?